pulp_ansible/app/viewsets.py

```diff
@@ -483,6 +889,59 @@ class AnsibleDistributionViewSet(DistributionViewSet):
     queryset = AnsibleDistribution.objects.all()
     serializer_class = AnsibleDistributionSerializer
 
+    queryset_filtering_required_permission = "ansible.view_ansibledistribution"
+
+    DEFAULT_ACCESS_POLICY = {
+        "statements": [
+            {
+                "action": ["list", "retrieve"],
+                "principal": "authenticated",
+                "effect": "allow",
+            },
+            {
+                "action": "create",
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": [
+                    "has_model_perms:ansible.add_ansibledistribution",
+                ],
+            },
+            {
+                "action": "destroy",
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": [
+                    "has_model_or_obj_perms:ansible.delete_ansibledistribution",
+                ],
+            },
+            {
+                "action": ["update", "partial_update"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": [
+                    "has_model_or_obj_perms:ansible.change_ansibledistribution",
+                ],
+            },
+        ],
+        "creation_hooks": [
+            {
+                "function": "add_roles_for_object_creator",
+                "parameters": {"roles": "ansible.ansibledistribution_owner"},
+            },
+        ],
+        "queryset_scoping": {"function": "scope_queryset"},
+    }
+
+    LOCKED_ROLES = {
+        "ansible.ansibledistribution_creator": ["ansible.add_ansibledistribution"],
+        "ansible.ansibledistribution_owner": [
+            "ansible.view_ansibledistribution",
+            "ansible.change_ansibledistribution",
+            "ansible.delete_ansibledistribution",
```

Comment:
Don't forget to add the `ansible.manage_roles_ansibledistribution` permission here after you add it to the model.